### ISSUE TITLE
[feature] consistent --json with output to stdout

### ISSUE THREAD

**Issue by mietzen** [conan-io/conan#6972]
[ x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
### What
I would like a consistent --json option which gives me a json string like:

```
conan inspect . --raw settings
```

does (even if the option is called --raw). For a search I now need to run:

```
conan search hello/master@user/testing -r artifacts -q 'os=Linux AND arch=x86_64' --json /tmp/test 2&>1 >/dev/null && cat /tmp/test
```

But I would like to run 

```
conan search hello/master@user/testing -r artifacts -q 'os=Linux AND arch=x86_64' --json
```

### Why

This would be really nice for CI/CD system like jenkins, this way we could easily parse the output directly to a groovy map, without disk IO.

### Where 

- info
- search
- profile
- remote
- inspect

**Comment by jgsogo** [comment]
Hi, @ngerke! 

Thanks for your suggestion. We are working on a new CLI and this is something we will consider. cc/ @czoido

**Comment by czoido** [comment]
Thanks a lot for the feedback @ngerke, definitely this is something we are going to take into account for conan 2.0 CLI.

**Comment by chausner-audeering** [comment]
I agree that the usefulness of the `--json` option is somewhat limited since it always writes to a file but normally you would want to avoid that when calling Conan from a script. Instead, the script would read the output of conan through a pipe.

I tried `--json /dev/stdout` on Linux which kind of works but Conan continues to write its normal output to stdout, as well, so the outputs get intermingled.

Some applications allow to pass "-" as filename to signify the output should be written to stdout instead of a file. That might be an option, or as @n-stone suggested, simply leaving out the path would also work.

**Comment by memsharded** [comment]
Conan 2.0 is getting closer:

- Formatters will output to stdout, to be able to pipe. 
- Files will be created with redirection
- App messages will be send to stderr to avoid polluting stdout output

**Comment by memsharded** [comment]
Released 2.0-beta6, hopefully last beta, implementing all these changes, so this can be closed
